pkg/registry/client_scope_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registry
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type RegistryScopeTestSuite struct {
+	TestRegistry
+}
+
+func (suite *RegistryScopeTestSuite) SetupSuite() {
+	// set registry use token auth
+	setup(&suite.TestRegistry, true, true, "token")
+
+}
+func (suite *RegistryScopeTestSuite) TearDownSuite() {
+	teardown(&suite.TestRegistry)
+	os.RemoveAll(suite.WorkspaceDir)
+}
+
+func (suite *RegistryScopeTestSuite) Test_1_Check_Push_Request_Scope() {
+
+	//set simple auth server to check the auth request scope
+	server := &http.Server{
```

Comment:
[`httptest.NewServer`](https://pkg.go.dev/net/http/httptest#NewServer) would be better ?

There is also [`NewOCIServer`](https://github.com/helm/helm/blob/5bab984cef25727762681b13e94bf5e687e6f3ac/pkg/repo/v1/repotest/server.go#L162) (not sure if this is entirely suitable for the tests here, but to try and avoid more bespoke test code)